src/controller/python/chip/commissioning/commissioning_flow_blocks.py

```diff
@@ -0,0 +1,182 @@
+#
+#    Copyright (c) 2023 Project CHIP Authors
+#    All rights reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+from chip import ChipDeviceCtrl
+from chip import commissioning
+from chip import clusters as Clusters
+
+import logging
+
+from . import pase
+
+
+class CommissioningFlowBlocks:
+    def __init__(self, devCtrl: ChipDeviceCtrl.ChipDeviceControllerBase, credential_provider: commissioning.CredentialProvider, logger: logging.Logger):
+        self._devCtrl = devCtrl
+        self._logger = logger
+        self._credential_provider = credential_provider
+
+    async def arm_failsafe(self, parameter: commissioning.PaseParameters, device: pase.Session):
+        response = await self._devCtrl.SendCommand(device.node_id, commissioning.ROOT_ENDPOINT_ID, Clusters.GeneralCommissioning.Commands.ArmFailSafe(
+            expiryLengthSeconds=parameter.failsafe_expiry_length_seconds
+        ))
+        if response.errorCode != 0:
+            raise commissioning.CommissionFailure(repr(response))
+
+    async def operational_credentials_commissioning(self, parameter: commissioning.Parameters, device: pase.Session):
+        self._logger.info("Getting Remote Device Info")
+        device_info = (await self._devCtrl.ReadAttribute(device.node_id, [
+            (commissioning.ROOT_ENDPOINT_ID, Clusters.BasicInformation.Attributes.VendorID),
+            (commissioning.ROOT_ENDPOINT_ID, Clusters.BasicInformation.Attributes.ProductID)], returnClusterObject=True))[commissioning.ROOT_ENDPOINT_ID][Clusters.BasicInformation]
+
+        self._logger.info("Getting AttestationNonce")
+        attestation_nonce = await self._credential_provider.get_attestation_nonce()
```

Comment:
Getting the nonces should be sequenced by the caller. For our usecases, we get the attestation and CSR nonces at the same time